pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -185,11 +204,24 @@ def on_pretrain_routine_start(self, trainer, pl_module):
         self.__resolve_ckpt_dir(trainer)
         self.save_function = trainer.save_checkpoint
 
+    def on_train_batch_end(self, trainer, pl_module, outputs, batch, batch_idx, dataloader_idx) -> None:
+        if self._should_skip_saving_checkpoint(trainer):
+            return
+        step = trainer.global_step
+        skip_step = self.every_n_steps < 1 or ((step + 1) % self.every_n_steps != 0)
+        if skip_step:
+            return
+        self.save_checkpoint(trainer, pl_module)
```

Comment:
nit: same structure as in `on_validation_end`

```suggestion
        if self.every_n_steps > 0 or ((step + 1) % self.every_n_steps == 1)
            self.save_checkpoint(trainer, pl_module)
```

Reply:
i left it this way as another thing I have in my personal model checkpoint callback is the option to save after a specified timedelta. i think this could be a general checkpointing feature and i'm happy to file a GH issue to track it.

this would also be applied on training batch end: we check whether we're past our timer, save, and reset

so we could save:
- after N batches frequency (during training phase)
- after T time (during training phase)
- after M epochs

and the change would be as simple as this:
```
       now = time.monotonic()
        time_interval = self.time_interval
        prev_time_check = self._prev_time_check
        skip_time = (
            time_interval is None
            or prev_time_check is None
            or (now - prev_time_check) < time_interval.total_seconds()
        )
        skip_time = trainer.training_type_plugin.broadcast(skip_time)
        if skip_step and skip_time:
            return
        if not skip_time:
            self._prev_time_check = now
        self.save_checkpoint(trainer, pl_module)
```

however i wanted to keep this to a separate PR . also in general i like the early returns to in order to avoid that extra level of nesting

Reply:
> after T time (during training phase)

That is an awesome feature! Would love for you to upstream it

> i like the early returns to in order to avoid that extra level of nesting

Me too, but in this case, it's one line after the return so not worth it having two exit points (IMO)